src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2190 @@
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, nn.Linear):
+            module.weight.data.normal_(mean=0.0, std=self.config.init_std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, (PatchTSMixerEncoder)):
+            module.gradient_checkpointing = value
+
+
+class PretrainHead(nn.Module):
+    """Pretraining head.
+
+    Args:
+        config (`PatchTSMixerConfig`, *mandatory*):
+            Configuration.
+    """
+
+    def __init__(self, config: PatchTSMixerConfig):
+        super().__init__()
+
+        num_patches = config.num_patches
+        num_features = config.num_features
+        input_size = config.input_size
+        patch_len = config.patch_len
+        head_dropout = config.head_dropout
+        mode = config.mode
+        self.mode = mode
+        self.patch_len = patch_len
+        self.input_size = input_size
+        self.num_patches = num_patches
+
+        if self.mode in ["common_channel", "mix_channel"]:
+            self.base_pt_block = nn.Sequential(
+                nn.Dropout(head_dropout),
+                nn.Linear(num_features, patch_len),
+            )
+        else:
+            self.base_pt_block = nn.Sequential(
+                nn.Dropout(head_dropout),
+                nn.Linear(num_features, patch_len * input_size),
+            )
+
+    def forward(self, hidden_features, target=None):
+        """
+        Args:
+            hidden_features (`torch.Tensor` of shape `(batch_size x num_patch x num_features)` in `flatten` mode
+                or `(batch_size x n_vars x num_patch x num_features)` in `common_channel`/`mix_channel` mode.): Input
+                hidden features.
+            target (`torch.Tensor`, *optional*):
+                Target tensor.
+
+        Returns:
+            `torch.Tensor` of shape `(batch_size x n_vars x num_patch x patch_len)`.
+        """
+
+        if self.mode == "flatten":
+            hidden_features = self.base_pt_block(
+                hidden_features
+            )  # hidden_features: [batch_size x num_patch x n_vars*patch_len]
+            hidden_features = torch.reshape(
+                hidden_features,
+                (
+                    hidden_features.shape[0],
+                    hidden_features.shape[1],
+                    self.patch_len,
+                    self.input_size,
+                ),
+            )  # [batch_size x num_patch x patch_len x n_vars]
+            hidden_features = hidden_features.permute(0, 3, 1, 2)  # [batch_size x nvars x num_patch  x patch_len]
+            return hidden_features
+        elif self.mode in ["common_channel", "mix_channel"]:
+            forecast = self.base_pt_block(hidden_features)  # [batch_size x n_vars x num_patch x patch_len]
+            return forecast
```

Comment:
What happens if self.mode is neither of the three? Can we raise here ? 
```suggestion
            return forecast
        else:
            raise ValueError(f"self.mode has to be one of ['flatten', 'common_channel', 'mix_channel'], but is {self.mode}.")
```

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/9112cbf3979d1aa8528882039ffd5ebad00e1259